app/gui/src/model/module/synchronized.rs

```diff
@@ -384,6 +423,10 @@ impl Module {
                     let notify_ls = self.notify_language_server(&summary.summary, &new_file, edits);
                     profiler::await_!(notify_ls, _profiler)
                 }
+                NotificationKind::Reloaded { summary, change } => {
+                    let notify_ls = self.notify_language_server(&summary, &new_file, vec![change]);
+                    profiler::await_!(notify_ls, _profiler)
+                }
```

Comment:
We notify the language server of the metadata changes that the parser introduces when loading the updated module content, not the changes performed by the language server. I can see how this is an awkward construction.

Would you prefer to have two separate notifications, one just for the synchronization with the language server and another for the parsing update? I think in that case the synchronization logic will have to be modified, as without the parsing step we do not have a `ParsedContentSummary` for the notification handler to keep track of.

Reply:
> Would you prefer to have two separate notifications, one just for the synchronization with the language server and another for the parsing update? I think in that case the synchronization logic will have to be modified, as without the parsing step we do not have a `ParsedContentSummary` for the notification handler to keep track of.

Well, as I think about it, it could stay this way, or we should call "notify_language_server" inside apply_text_change_from_ls (after updating our content). I slightly prefer the latter solution. Add a comment with explanation.